From 10052fbab5df6771dca56541656258217fb1c2c8 Mon Sep 17 00:00:00 2001
From: Shahed Shaikh <shahedx.shaikh@intel.com>
Date: Sun, 12 Jul 2020 13:06:30 +0800
Subject: [PATCH] DRV_DPM_SW-329: DPM minor improvements

- in dp_get_netif_subifid, change all pr_err to DP_DEBUG(DP_DBG_FLAG_DBG,...)
- in dp_alloc_port_private(), call get_dp_port_info() only when port_id is
  is provided by caller in case of port registration.
- print skb->priority in dp_xmit()

Signed-off-by: Shahed Shaikh <shahedx.shaikh@intel.com>
---
 drivers/net/datapath/dpm/datapath.h     |  3 ---
 drivers/net/datapath/dpm/datapath_api.c | 11 +++++++----
 drivers/net/datapath/dpm/datapath_tx.c  |  4 ++--
 3 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/net/datapath/dpm/datapath.h b/drivers/net/datapath/dpm/datapath.h
index 3128769b5d66..62b0c1c04da3 100644
--- a/drivers/net/datapath/dpm/datapath.h
+++ b/drivers/net/datapath/dpm/datapath.h
@@ -109,9 +109,6 @@ static inline bool is_invalid_inst(unsigned int inst)
 #define IFNAMSIZ 16
 #define DP_MAX_HW_CAP 4
 
-#if (!IS_ENABLED(CONFIG_PRX300_CQM))
-#define DP_SPIN_LOCK
-#endif
 #ifdef DP_SPIN_LOCK
 #define DP_LOCK_T spinlock_t
 #define DP_LOCK_INIT(lock) spin_lock_init(lock)
diff --git a/drivers/net/datapath/dpm/datapath_api.c b/drivers/net/datapath/dpm/datapath_api.c
index 1d37d8400e3e..047a3e7106ce 100644
--- a/drivers/net/datapath/dpm/datapath_api.c
+++ b/drivers/net/datapath/dpm/datapath_api.c
@@ -426,9 +426,9 @@ static int32_t dp_alloc_port_private(int inst,
 	dp_info = get_dp_prop_info(inst);
 	cbm_data.dp_inst = inst;
 	cbm_data.cbm_inst = dp_port_prop[inst].cbm_inst;
-	port = get_dp_port_info(inst, port_id);
 
 	if (flags & DP_F_DEREGISTER) {	/*De-register */
+		port = get_dp_port_info(inst, port_id);
 		if (port->status != PORT_ALLOCATED) {
 			pr_err("No Deallocate for module %s w/o deregistered\n",
 			       owner->name);
@@ -450,6 +450,7 @@ static int32_t dp_alloc_port_private(int inst,
 		return DP_SUCCESS;
 	}
 	if (port_id) { /*with specified port_id */
+		port = get_dp_port_info(inst, port_id);
 		if (port->status != PORT_FREE) {
 			pr_err("%s %s(%s %d) fail: port %d used by %s %d\n",
 			       "module", owner->name,
@@ -1566,7 +1567,8 @@ int32_t dp_get_netif_subifid(struct net_device *netif, struct sk_buff *skb,
 	int res = DP_FAILURE;
 
 	if (unlikely(!dp_init_ok)) {
-		pr_err("%s failed: datapath not initialized yet\n", __func__);
+		 DP_DEBUG(DP_DBG_FLAG_DBG,
+			    "%s failed: datapath not initialized yet\n", __func__);
 		return res;
 	}
 
@@ -1575,7 +1577,7 @@ int32_t dp_get_netif_subifid(struct net_device *netif, struct sk_buff *skb,
 	rcu_read_lock_bh();
 	dp_subif = dp_subif_lookup_safe(&dp_subif_list[idx], netif, subif_data);
 	if (!dp_subif) {
-		pr_err("Failed dp_subif_lookup: %s\n",
+		DP_DEBUG(DP_DBG_FLAG_DBG, "Failed dp_subif_lookup: %s\n",
 		       netif ? netif->name : "NULL");
 		rcu_read_unlock_bh();
 		return res;
@@ -1596,7 +1598,8 @@ int32_t dp_get_netif_subifid(struct net_device *netif, struct sk_buff *skb,
 		res = subifid_fn_t(netif, skb, subif_data, dst_mac, subif,
 				   flags);
 		if (res != 0)
-			pr_err("get_netif_subifid callback function failed\n");
+			DP_DEBUG(DP_DBG_FLAG_DBG,
+				   "get_netif_subifid callback function failed\n");
 	} else {
 		res = DP_SUCCESS;
 	}
diff --git a/drivers/net/datapath/dpm/datapath_tx.c b/drivers/net/datapath/dpm/datapath_tx.c
index a8d5497df6ef..475d96f8044a 100644
--- a/drivers/net/datapath/dpm/datapath_tx.c
+++ b/drivers/net/datapath/dpm/datapath_tx.c
@@ -113,8 +113,8 @@ void dp_tx_dbg(char *title, struct sk_buff *skb, s32 ep, s32 len, u32 flags,
 	;
 #endif
 
-	DP_DEBUG(DP_DBG_FLAG_DUMP_TX, "flags=0x%x skb->len=%d\n",
-		 flags, skb->len);
+	DP_DEBUG(DP_DBG_FLAG_DUMP_TX, "flags=0x%x skb->len=%d skb->priority=%d\n",
+		 flags, skb->len, skb->priority);
 	DP_DEBUG(DP_DBG_FLAG_DUMP_TX,
 		 "skb->data=0x%px with pmac hdr size=%zu\n", skb->data,
 		 sizeof(struct pmac_tx_hdr));
